### ISSUE TITLE
Explore automated accessibility testing for Tutor, cnx, and openstax.org

### ISSUE THREAD

**Issue by ambersdai** [openstax/cnx#120]
## Overview
One of the places we should run automated tests are as part of unit testing (others include integration testing)

Some options for running them during unit testing include:

- write a separate set of tests that spin up the app and click around (harder to keep up-to-date)
- include a11y reporting _during_ unit testing... maybe as a plugin to Jest or something… sort of like code coverage (harder to run since many a11y libs require a browser, not jsdom) See [nickcolley/jest-axe](https://github.com/nickcolley/jest-axe)
- run a11y tests on the `.snip `files after unit tests run (harder to run since many a11y libs require the `<body>` and might not have enough context)

Also, since our code does not pass WCAG guidelines it might be too difficult to fix everything at once. Instead, we could do something like code-coverage... just ensure that we have fewer failures than before. One way to do this is to have a TXT file that lists all of the WCAG guidelines and then, when we fail a guideline we could reduce coverage of that line (the WCAG guideline that failed)

Related cards:
- a11y: Run AXE tools on each Page on Travis

[Slack: Decision to include in Unicorn
](https://openstax.slack.com/archives/C836DQHE1/p1520371792000148)
[Slack: Idea for making reports
](https://openstax.slack.com/archives/C4PJYP0LF/p1520278416000622)

## PRs
https://github.com/openstax/tutor-js/pull/2168

## Test Plan
1. Figure out options for running the tests

- try incorporating an a11y testing library that runs when Jest Runs
- if that does not work, try incorporating an a11y testing library that runs on Jest snippet files
- if that does not work, try writing new tests that start up a browser (using puppeteer) and run an a11y testing library on each of the pages in the app

2. Figure out how to report
- try making the coverage file
- try just failing if a WCAG guideline fails

## Acceptance Criteria

- We found a way to run tests
- We found a way to report the findings and cause Travis/codecov to show the results in some way that is failed/reported to a developer
